templates/cicd/pipeline_cfn.yml

```diff
@@ -233,29 +232,43 @@ Resources:
       RoleArn: !GetAtt PipelineRole.Arn
       Name: !Ref AWS::StackName
       Stages:
+        {{- if eq .Source.ProviderName "GitHub"}}
         - Name: Source
           Actions:
             - Name: SourceCodeFor-{{$.AppName}}
               ActionTypeId:
                 Category: Source
-                # TODO: #273 For CodeCommit, this needs to be AWS. Let's do that later
                 Owner: ThirdParty
                 Version: 1
-                Provider: {{.Source.ProviderName}}
+                Provider: GitHub
               Configuration:
-                # TODO: #273 For now this set of configurations is GitHub-specific,
-                # change it to be more generic
                 Owner: {{$.Source.Owner}}
                 Repo: {{$.Source.Repository}}
```

Comment:
Do these fields come from `deploy.GithubSource` or `manifest.GithubProperties`, or somewhere else? In those structs the field is `RepositoryURL`, not `Repository`--is that correct?

Reply:
Good question-- this is confusing.
Repository is the method (`func (s *GitHubSource) Repository() (string, error)`)  in the deploy package that returns the repo of a GitHubSource. For this field, we want just the repo name, not the whole url.

Reply:
Interesting, so Go will call the method named there. I talked about this in another comment where I think we can combine a base struct with a public ProviderName method and generator function. It's not super necessary though so if you think it's too much bother feel free to 🚀